### PR TITLE
tikv-ctl: implement drop unapplied log for v2

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref https://github.com/tikv/tikv/issues/14654

What's Changed:

```commit-message
implement drop unapplied log for v2
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/server/debug2.rs [hunk 1]
```diff
@@ -472,6 +472,73 @@ impl<ER: RaftEngine> DebuggerImplV2<ER> {
...
+            }
+
+            let old_raft_local_state = region_state.raft_local_state.ok_or_else(|| {
+                Error::Other(format!("No RaftLocalState found for region {}", region_id).into())
+            })?;
+            let old_raft_apply_state = region_state.raft_apply_state.ok_or_else(|| {
+                Error::Other(format!("No RaftApplyState found for region {}", region_id).into())
+            })?;
+
+            let applied_index = old_raft_apply_state.applied_index;
```
Comment (overvenus) [inline src/server/debug2.rs:497]: Should we use persisted index?

Comment (SpadeA-Tang) [reply src/server/debug2.rs:497]: Good catch.

File: src/server/debug2.rs [hunk 2]
```diff
@@ -472,6 +472,73 @@ impl<ER: RaftEngine> DebuggerImplV2<ER> {
...
+                "region_id" => region_id,
+                "old_raft_local_state" => ?old_raft_local_state,
+                "new_raft_local_state" => ?new_raft_local_state,
+                "old_raft_apply_state" => ?old_raft_apply_state,
+                "new_raft_apply_state" => ?new_raft_apply_state,
+            );
+
+            // flush the changes
+            let mut lb = raft_engine.log_batch(10);
+            box_try!(lb.put_apply_state(region_id, applied_index, &new_raft_apply_state));
```
Comment (overvenus) [inline src/server/debug2.rs:525]: Should we update persisted index too?

**Review by bufferflies — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**